src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -25,16 +28,32 @@
      */
     public CommandBox(CommandExecutor commandExecutor) {
         super(FXML);
+        userInputHistory = new UserInputHistory();
         this.commandExecutor = commandExecutor;
         // calls #setStyleToDefault() whenever there is a change to the text of the command box.
         commandTextField.textProperty().addListener((unused1, unused2, unused3) -> setStyleToDefault());
     }
 
+    /**
+     * Handles the Up or Down key pressed event.
+     */
+    @FXML
+    private void handleKeyPressed(KeyEvent event) {
+        if (event.getCode() == KeyCode.UP) {
+            String prevInput = userInputHistory.getPreviousInput();
+            commandTextField.setText(prevInput);
+        } else if (event.getCode() == KeyCode.DOWN) {
+            String nextInput = userInputHistory.getNextInput();
+            commandTextField.setText(nextInput);
+        }
+    }
+
     /**
      * Handles the Enter button pressed event.
      */
     @FXML
     private void handleCommandEntered() {
+        userInputHistory.addToHistory(commandTextField.getText());
         try {
```

Comment:
you might want to consider only adding the text if its not blank?

Reply:
Yeah should've considered that earlier.